diffkemp/semdiff/function_diff.py

```diff
@@ -159,7 +159,8 @@ def functions_semdiff(first, second, fun_first, fun_second, config):
 
 def functions_diff(mod_first, mod_second,
                    fun_first, fun_second,
-                   glob_var, config):
+                   glob_var, config,
+                   cache=None, simpll_cache=None):
```

Comment:
The naming here is a bit confusing - what is the difference between `cache` and `simpll_cache`? And how does `graph` come into it? We should probably distinguish between a comparison graph (a structure that holds all results obtained so far) and a cache (a set of files generated from the graph).

Reply:
`cache` is the graph from the previous comparison, `graph` is the current graph (which represents the result of SimpLL before merging and all results up to this KABI symbol after the merging), `simpll_cache` is passed *to* SimpLL (while `graph` is passed *from* SimpLL) - the naming is indeed quite confusing, I should definitely improve it.